lib/dry/view/controller.rb

```diff
@@ -24,6 +25,16 @@ class Controller
       setting :layout, false
       setting :template
       setting :default_format, :html
+      setting :renderer_options, DEFAULT_RENDERER_OPTIONS do |v|
+        unless v.is_a?(::Hash)
+          if v.respond_to?(:to_h) then v = v.to_h
+          elsif v.respond_to?(:to_hash) then v = v.to_hash
+          else raise ArgumentError, "Expected a Hash for +renderer_options+ but recevied a #{v.class}"
+          end
+        end
+
+        DEFAULT_RENDERER_OPTIONS.merge(v)
```

Comment:
Shouldn't we `freeze` the new merged value? 

I don't know is totally necessary but since we are doing in the declaration of `DEFAULT_RENDERER_OPTIONS` I felt the need to ask it 😄

Reply:
Done! I think it makes sense to be consistently safe from mutation.